canonical-web-and-design/maas-ui
# refactor(ui): maintain disk/partition hierarchy in storage tables

## Done

- Refactored Available/Used disks and partitions tables to no longer use the `separateStorageData` util, which flattened the data hierarchy.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a ready or allocated machine
- Check that the data in the available and used disks and partitions tables remains the same
- Check that changing the storage layout updates the tables correctly
- Check that adding a partition to a disk still works as before

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2282
Fixes canonical-web-and-design/MAAS-squad#2283
